drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,590 @@
+#define GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH             (0x3U)
+
+static const char *const ra8_pwm_errmsg[] = {
+	[FSP_ERR_ASSERTION] = "[FSP_ERR_ASSERTION] A required input pointer is NULL "
+			      "OR the source divider is invalid",
+	[FSP_ERR_ALREADY_OPEN] = "[FSP_ERR_ALREADY_OPEN] Module is already open",
+	[FSP_ERR_IRQ_BSP_DISABLED] =
+		"[FSP_ERR_IRQ_BSP_DISABLED] ISR must be enabled to use callback",
+	[FSP_ERR_INVALID_MODE] =
+		"[FSP_ERR_INVALID_MODE] Triangle wave PWM is only supported if "
+		"GPT_CFG_OUTPUT_SUPPORT_ENABLE is 2 "
+		"OR Selected channel does not support external count sources "
+		"OR External and event count sources not are available in this mode",
+	[FSP_ERR_NOT_OPEN] = "[FSP_ERR_NOT_OPEN] The instance is not opened",
+	[FSP_ERR_INVALID_ARGUMENT] = "[FSP_ERR_INVALID_ARGUMENT] Duty cycle is larger than period",
+	[FSP_ERR_UNSUPPORTED] = "[FSP_ERR_UNSUPPORTED] PLL GPT_CFG_OUTPUT_SUPPORT_ENABLE is 0"};
+
+#define ZEPHYR_ERROR_LOG(err)                                                                      \
+	{                                                                                          \
+		if (err == FSP_SUCCESS) {                                                          \
+			(void)0;                                                                   \
+		} else {                                                                           \
+			LOG_ERR("%s: %s", __func__, ra8_pwm_errmsg[err]);                          \
+		}                                                                                  \
+	}
+struct pwm_ra8_capture_data {
+	pwm_capture_callback_handler_t callback;
+	void *user_data;
+	uint64_t period;
+	uint64_t pulse;
+	bool is_pulse_capture;
+	bool is_busy;
+	uint32_t overflows;
+	bool continuous;
+};
+
+struct pwm_ra8_data {
+	gpt_instance_ctrl_t fsp_ctrl;
+	timer_cfg_t fsp_cfg;
+	gpt_extended_cfg_t extend_cfg;
+	uint16_t capture_a_event;
+	uint16_t overflow_event;
+
+#ifdef CONFIG_PWM_CAPTURE
+	struct pwm_ra8_capture_data capture;
+#endif /* CONFIG_PWM_CAPTURE */
+};
+
+struct pwm_ra8_config {
+	const struct device *clock_dev;
+	struct clock_control_ra_subsys_cfg clock_subsys;
+	const struct pinctrl_dev_config *pincfg;
+};
+
+static uint32_t pwm_ra8_gtior_calculate(timer_cfg_t const *const p_cfg,
+					gpt_pin_level_t const stop_level)
+{
+	/* The stop level is used as both the initial level and the stop level. */
+	uint32_t gtior = R_GPT0_GTIOR_OAE_Msk | ((uint32_t)stop_level << R_GPT0_GTIOR_OADFLT_Pos) |
+			 ((uint32_t)stop_level << GPT_PRV_GTIOR_INITIAL_LEVEL_BIT);
+
+	uint32_t gtion = GPT_PRV_GTIO_LOW_COMPARE_MATCH_HIGH_CYCLE_END;
+
+	if (TIMER_MODE_PWM == p_cfg->mode) {
+		/* Use default: GTIOn is high at cycle end, then low at compare match. */
+	}
+
+#if GPT_PRV_EXTRA_FEATURES_ENABLED == GPT_CFG_OUTPUT_SUPPORT_ENABLE
+	else if (p_cfg->mode >= TIMER_MODE_TRIANGLE_WAVE_SYMMETRIC_PWM) {
+		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
+	}
+#endif
```

Comment:
this should really be a `switch (p_cfg->mode)`

Reply:
sure

Reply:
Actually, would it be possible to do

```c
	} else if (p_cfg->mode >= TIMER_MODE_TRIANGLE_WAVE_SYMMETRIC_PWM) {
#if GPT_PRV_EXTRA_FEATURES_ENABLED == GPT_CFG_OUTPUT_SUPPORT_ENABLE
		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
#endif
	} else if {
```

instead? Looking for a way of not splitting the if-else

Reply:
how about: could you make a separate functoin that gets `p_cfg->mode` as an input and returns the `gtion` value? Then you can have intermediate returns and you should be able to avoid the split if-else I think

Reply:
@fabiobaltieri:  Sorry for the late response.
Actually we use only TIMER_MODE_PWM on PWM driver with our GPT hardware. For some reason, we defined this function with multiple modes, but to avoid confusion, I removed the other, only define1 value `gtion` for TIMER_MODE_PWM here. Thanks again for your comments.